File: leet-code/191. Number of 1 Bits/NumberOf1Bits191_3.java
package com.example;

public class NumberOf1Bits191_3 {
    public int hammingWeight(int n) {
        int count = 0;
        while (n > 0) {
            if (n % 2 == 1) {
                count += 1;
            }
            n /= 2;
        }
        return count;
    }
}

File: leet-code/191. Number of 1 Bits/NumberOf1Bits191_3.kt
package com.example

class NumberOf1Bits191_3 {
    fun hammingWeight(n: Int): Int {
        var num = n
        var count = 0
        while (num > 0) {
            if (num % 2 == 1) {
                count++
            }
            num /= 2
        }
        return count
    }
}
